pkg/network/dns_stats.go

```diff
@@ -113,7 +119,12 @@ func (d *dnsStatKeeper) ProcessPacketInfo(info dnsPacketInfo, ts time.Time) {
 	}
 	stats, ok := allStats[start.question]
 	if !ok {
+		if d.numStats >= d.maxStats {
+			d.droppedStats++
+			return
+		}
 		stats.DNSCountByRcode = make(map[uint32]uint32)
+		d.numStats++
```

Comment:
Can we use `len(allStats)` instead of keeping track of the stat size in `d.numStats` ?

Reply:
Then we will have to loop over and sum `len(allStats)` for all `DNSKey`s . Currently, `stats` is organized in the following way:
```
|              ___ bar.com
|             |
|---dnsKey1---|
|             |___ foo.com
|
|              ___ baz.com
|             |
|---dnsKey2---|
|             |___ foo.com
```